src/artifact_providers/github.ts

```diff
@@ -93,35 +102,37 @@ export class GithubArtifactProvider extends BaseArtifactProvider {
       artifact => artifact.name === revision
     );
 
-    if (foundArtifacts.length !== 1) {
+    if (foundArtifacts.length === 0) {
+      if (artifactResponse.total_count > per_page) {
```

Comment:
Looks like this have a potential of entering the infinite loop if the condition holds initially, because both `total_count` and `per_page` don't change.